docs/generating-a-provider.md

```diff
@@ -0,0 +1,391 @@
+       ```
+
+       Run:
+       ```bash
+       go mod tidy
+       ```
+
+       Please note, we are temporarily using a [fork] of
+       [terraform-provider-github] repo as a workaround to [this issue].
+
+    4. If your provider uses an old version (<v2) of [terraform-plugin-sdk],
+       convert resource map to v2 schema as follows (uncomment related section):
+
+       ```go
+       import (
+           "github.com/crossplane-contrib/terrajet/pkg/types/conversion"
+       )
+       
+       resourceMap := conversion.GetV2ResourceMap(tf.Provider())
+       ```
+
+       And in `go.mod` file, set the following `replace directive`
+       (uncomment related section):
+
+       ```
+       github.com/hashicorp/terraform-plugin-sdk => github.com/turkenh/terraform-plugin-sdk v1.17.2-patch1
+       ```
+
+       Run:
+       ```bash
+       go mod tidy
+       ```
+
+5. Implement `ProviderConfig` logic. In `provider-tf-template`, there is already
+   a boilerplate code in file `internal/clients/${ProviderNameLower}.go` which
+   takes care of properly fetching secret data referenced from `ProviderConfig`
+   resource.
+
+   For our GitHub provider, we need to check [Terraform documentation for provider
+   configuration] and provide the keys there:
+
+   ```go
+   const (
+     keyBaseURL = "base_url"
+     keyOwner = "owner"
+     keyToken = "token"
+
+     // GitHub credentials environment variable names
+     envToken = "GITHUB_TOKEN"
+   )
+
+   func TerraformSetupBuilder(version, providerSource, providerVersion string) terraform.SetupFn {
+     ...
+     // set provider configuration
+     ps.Configuration = map[string]interface{}{}
+     if v, ok := githubCreds[keyBaseURL]; ok {
+         ps.Configuration[keyBaseURL] = v
+     }
+     if v, ok := githubCreds[keyOwner]; ok {
+         ps.Configuration[keyOwner] = v
+     }
+     // set environment variables for sensitive provider configuration
+     ps.Env = []string{
+         fmt.Sprintf(fmtEnvVar, envToken, githubCreds[keyToken]),
+     }
+     return ps, nil
+   }
+   ```
+
+6. Before generating all resources that the provider has, let's go step by step
+   and only start with generating `github_repository` and `github_branch`
+   resources.
+
+   To limit the resources to be generated, we need to provide an include list
+   option with `tjconfig.WithIncludeList` in file `config/provider.go`:
+
+   ```go
+   pc := tjconfig.NewProvider(resourceMap, resourcePrefix, modulePath,
+       tjconfig.WithDefaultResourceFn(defaultResourceFn),
+       tjconfig.WithIncludeList([]string{
+           "github_repository$",
+           "github_branch$",
+       }))
+   ```
+
+7. Finally, we would need to add some custom configurations for these two
+   resources as follows:
+
+   ```bash
+   # Create custom configuration directory for whole repository group
+   mkdir config/repository
+   # Create custom configuration directory for whole branch group
+   mkdir config/branch
+   ```
+
+   ```bash
```

Comment:
```suggestion
   ```go
```

Because they won't write the exact code, seeing it in command format won't really help as much. So, I'd suggest having YAMLs and Go snippets as is to get syntax highlighting and get users to copy-paste them in their editors instead of running commands. The same doesn't go for the generic find&replace commands we have at the top for example, because they will run them regardless of the provider they're generating.

Reply:
these files don't exist at this point and I found this to be more convenient while following the guide, instead of creating directory, file and copy/pasting file content to an editor.